### PR TITLE
feat: Add function finish_with_stream_position to the IpcStreamWriter

### PR DESCRIPTION
**Pull request by SimonFlarion**

### CODE REVIEW

File: crates/polars-io/src/ipc/ipc_stream.rs [hunk 1]
```diff
@@ -359,6 +359,18 @@ impl<W: Write> IpcStreamWriter<W> {
...
 }
 
+impl<W: Write + Seek> IpcStreamWriter<W> {
+    pub fn finish_with_stream_position(mut self, df: &mut DataFrame) -> PolarsResult<usize> {
+        self.finish(df)?;
+
+        if let Ok(pos) = self.writer.stream_position() {
+            return Ok(pos as usize);
+        } else {
+            polars_bail!(ComputeError: "Can't get position in writer")
```
Comment (eshed-flarion) [inline crates/polars-io/src/ipc/ipc_stream.rs]: I think the behaviour should be the same for `IpcStreamBatchedWriter` and `IpcStreamWriter`, either both bail if `stream_position` fails or both fallback to bytes written.

File: crates/polars-io/src/ipc/ipc_stream.rs [hunk 2]
```diff
@@ -359,6 +359,18 @@ impl<W: Write> IpcStreamWriter<W> {
     }
 }
 
+impl<W: Write + Seek> IpcStreamWriter<W> {
+    pub fn finish_with_stream_position(mut self, df: &mut DataFrame) -> PolarsResult<usize> {
+        self.finish(df)?;
+
+        if let Ok(pos) = self.writer.stream_position() {
+            return Ok(pos as usize);
```
Comment (EmilyFlarionIO) [inline crates/polars-io/src/ipc/ipc_stream.rs]: Can remove return and have polars_err!

File: crates/polars-io/src/ipc/ipc_stream.rs (hunk 1 above)
Comment (SimonFlarion) [reply crates/polars-io/src/ipc/ipc_stream.rs]: Done

File: crates/polars-io/src/ipc/ipc_stream.rs (hunk 2 above)
Comment (SimonFlarion) [reply crates/polars-io/src/ipc/ipc_stream.rs]: Done

**Review by EmilyFlarionIO — APPROVED** [review]
*hacker voice* lgtm

**Pull request merged**